gtsfm/frontend/verifier/ransac.py

```diff
@@ -21,68 +23,91 @@
+            use_intrinsics_in_verification (optional): Flag to perform keypoint normalization and compute the essential
+                                                       matrix instead of fundamental matrix. This should be preferred
+                                                       when the exact intrinsics are known as opposed to approximating
+                                                       them from exif data. Defaults to False.
 
         Returns:
             Estimated rotation i2Ri1, or None if it cannot be estimated.
             Estimated unit translation i2Ui1, or None if it cannot be estimated.
-            Indices of verified correspondences, of shape (N, 2) with N <= N3.
-                These indices are subset of match_indices.
+            Indices of verified correspondences, of shape (N, 2) with N <= N3. These are subset of match_indices.
         """
         verified_indices = np.array([], dtype=np.uint32)
 
         # check if we don't have the minimum number of points
-        if match_indices.shape[0] < self.min_pts:
+        if match_indices.shape[0] < self.min_pts_e_matrix:
             logger.info("No match indices were provided to the verifier, returning early with None output")
             return None, None, verified_indices
 
         uv_norm_i1 = feature_utils.normalize_coordinates(keypoints_i1.coordinates, camera_intrinsics_i1)
         uv_norm_i2 = feature_utils.normalize_coordinates(keypoints_i2.coordinates, camera_intrinsics_i2)
         K = np.eye(3)
 
-        i2Ei1, inlier_mask = cv2.findEssentialMat(
-            uv_norm_i1[match_indices[:, 0]],
-            uv_norm_i2[match_indices[:, 1]],
-            K,
-            method=cv2.RANSAC,
-            threshold=NORMALIZED_COORD_RANSAC_THRESH,
-            prob=DEFAULT_RANSAC_SUCCESS_PROB,
-        )
+        if use_intrinsics_in_verification:
+            i2Ei1, inlier_mask = cv2.findEssentialMat(
+                uv_norm_i1[match_indices[:, 0]],
+                uv_norm_i2[match_indices[:, 1]],
+                K,
+                method=cv2.RANSAC,  # cv2.USAC_DEFAULT # cv2.RANSAC
+                threshold=NORMALIZED_COORD_RANSAC_THRESH,
+                prob=DEFAULT_RANSAC_SUCCESS_PROB,
+            )
+        else:
+            # TODO: debug what is happening and try to fix it
+            if match_indices.shape[0] < self.min_pts_f_matrix:
+                return None, None, verified_indices
+
+            i2Fi1, inlier_mask = cv2.findFundamentalMat(
+                keypoints_i1.extract_indices(match_indices[:, 0]).coordinates,
+                keypoints_i2.extract_indices(match_indices[:, 1]).coordinates,
+                method=cv2.FM_RANSAC,  # cv2.USAC_FM_8PTS # cv2.FM_RANSAC
+                ransacReprojThreshold=PIXEL_COORD_RANSAC_THRESH,
+                confidence=DEFAULT_RANSAC_SUCCESS_PROB,
+                maxIters=10000,
+            )
+
+            if i2Fi1 is None:
```

Comment:
when does cv2 return None here?

Reply:
removed it. I could not figure out why I added the check and hence cannot reproduce it.